### PR TITLE
Refactor pylance reusable pieces to only depend upon protocol types

### PR DESCRIPTION
**Pull request by rchiodo**
For https://github.com/microsoft/vscode-jupyter/issues/8467

### CODE REVIEW

File: src/protocol-only/notebookConcatDocument.ts [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Licensed under the MIT License.
 
 'use strict';
-import * as vscode from 'vscode';
-import * as protocol from 'vscode-languageclient/node';
+import * as vscodeUri from 'vscode-uri';
```
Comment (rchiodo) [inline src/protocol-only/notebookConcatDocument.ts:5]: @Heejaechang @bschnurr this file should now be not using any of the vscode types.

File: src/protocol-only/notebookConverter.ts [hunk 2]
```diff
@@ -0,0 +1,952 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as os from 'os';
```
Comment (rchiodo) [inline src/protocol-only/notebookConverter.ts:3]: Same with this file. Should be vscode free.

File: src/protocol-only/notebookConcatDocument.ts [hunk 3]
```diff
@@ -2,8 +2,8 @@
...
+import * as protocol from 'vscode-languageclient';
```
Comment (bschnurr) [inline src/protocol-only/notebookConcatDocument.ts:6]: Not sure if it helps but we get all the types for TextDocument, Range, Position from `vscode-languageserver-types` lib

Comment (rchiodo) [reply src/protocol-only/notebookConcatDocument.ts:6]: Oh thanks. I'll try that and see what happens. Should be the same I think.

Comment (rchiodo) [reply src/protocol-only/notebookConcatDocument.ts:6]: Didn't work, but 'vscode-languageserver-protocol' seems to work. I'll switch to that one.

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**